source/eventHandler.py

```diff
@@ -143,6 +143,9 @@ def executeEvent(eventName,obj,**kwargs):
 	@param kwargs: Additional event parameters as keyword arguments.
 	"""
 	try:
+		# Allow NVDAObjects to redirect focus events to another object of their choosing.
+		if eventName=="gainFocus" and obj.focusRedirect:
+			obj=obj.focusRedirect

```

Comment:
focusRedirect is used in the powerpnt appModule. We might have to make sure that this does not break. Having said that, I really like this being handled on the events level!